fenicsadapter/fenicsadapter.py

```diff
@@ -259,501 +38,308 @@ def __init__(self, adapter_config_filename='precice-adapter-config.json',
         self._interface = precice.Interface(self._solver_name, self._config.get_config_file_name(), 0, 1)
         self._dimensions = self._interface.get_dimensions()
```

Comment:
We might consider removing this member and always call `self._interface.get_dimensions()`.

Reply:
The variable `self._dimensions` is used multiple times in each time step. Is it advisable to call a preCICE API function multiple times within on time step? The current thinking is that the API function needs to be called only once and the adapter then uses the locally stored value till the end of the coupling.

Reply:
This is clearly a conflict runtime vs. readability. Less members decrease complexity, but calling the API too often, might indeed be harmful. Can you do a quick check to find out how time it will cost if we do not store the dimensions in the adapter, but always go through the preCICE API, if we need the dimensions?

I would suggest: If no real runtime problem occurs favour readability. 

Remember: FEniCS' solve is probably much more expensive and we do not really have to worry about this additional API call, at all.

Reply:
I did a time study with respect to problem size with reference to all preCICE API calls which return single values. These calls are: `get_dimensions`, `get_mesh_id`, `get_write_data_id`, `get_read_data_id`. I compared the Global Time when these calls are made each time the respective value is required vs. the Adapter calls these functions once and stores them in local variables which are reusable.

![timestudy](https://user-images.githubusercontent.com/44898158/81546435-a7479180-937a-11ea-9a79-636e5f27f71d.png)

No significant time difference is seen for both scenarios and both scenarios scale in almost the same way for increasing problem size.

Reply:
Thanks for the study! Then I would suggest to get rid of `self._dimensions` and always go over the API function `interface.get_dimensions()`